deepchem/data/tests/test_datasets.py

```diff
@@ -395,6 +397,109 @@ def test_disk_iterate_batch_size(self):
       batch_sizes.append(len(X))
     self.assertEqual([3, 3, 3, 1], batch_sizes)
 
+  def test_disk_iterate_batch(self):
```

Comment:
Could you break this into multiple simpler tests? It's hard to follow along with the test given the length.

Reply:
All it really does is create a dataset, then iterate over it in both deterministic and nondeterministic order and make sure each one gives the right results.  It repeats that a bunch of times for random patterns of shard sizes.  It could maybe be simplified a bit, but I don't think it makes sense to break it into multiple tests.